signalfx/splunk-otel-js-browser
# Move span attribute setting into a SpanProcessor

# Description

Remove custom startSpan in WebTracer and move its attribute setting into SpanProcessor. This is a lot less hacky way to do it using an already existing synchronous hook.



Current method was also causing potential performance issues which turned into stack overflow error:

Parent `WebTracerProvider` implementation returns the same Tracer instance when it's called with same parameters. So `trace.getTracer('default')  === trace.getTracer('default')`. But since the override always replaces the current startSpan with patched startSpan, on 2nd call it would get already patched startSpan as original.

So do `trace.getTracer('default')` 5 times, end up with `startSpan === wrappedStartSpan -> wrappedStartSpan -> wrappedStartSpan -> wrappedStartSpan -> wrappedStartSpan -> origStartSpan`

Do `trace.getTracer('default')` 4000+ times and you end up with stack overflow

Since re-setting an attribute with same key just overwrites it, it wasn't a noticeable bug.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore or internal change (changes not visible to the consumers of the package)

# How has this been tested?

- Manual testing
- (Changed &) Ran unit tests (to confirm no differences)
- Ran integration tests (to confirm no differences)
